src/platform/tests/Bridge/HuggingFace/ApiClientTest.php

```diff
@@ -0,0 +1,120 @@
+
+        $this->assertCount(3, $models);
+        $this->assertInstanceOf(Model::class, $models[0]);
+        $this->assertSame('model-1', $models[0]->getName());
+        $this->assertSame('model-2', $models[1]->getName());
+        $this->assertSame('model-3', $models[2]->getName());
+    }
+
+    #[TestDox('Handles null provider and task parameters correctly')]
+    public function testModelsWithNullProviderAndTask()
+    {
+        $responseData = [
+            ['id' => 'model-1'],
+            ['id' => 'model-2'],
+        ];
+
+        $httpClient = new MockHttpClient(new JsonMockResponse($responseData));
+        $apiClient = new ApiClient($httpClient);
+
+        $models = $apiClient->models(null, null);
+
+        $this->assertCount(2, $models);
+        $this->assertInstanceOf(Model::class, $models[0]);
+        $this->assertSame('model-1', $models[0]->getName());
+        $this->assertSame('model-2', $models[1]->getName());
+    }
+
+    #[TestDox('Returns empty array when API responds with no models')]
+    public function testModelsWithEmptyResponse()
+    {
+        $responseData = [];
+
+        $httpClient = new MockHttpClient(new JsonMockResponse($responseData));
+        $apiClient = new ApiClient($httpClient);
+
+        $models = $apiClient->models('test-provider', 'text-generation');
+
+        $this->assertCount(0, $models);
+        $this->assertIsArray($models);
+    }
+
+    #[TestDox('Sends correct HTTP request with provider and task parameters')]
+    public function testModelsRequestParameters()
+    {
+        $httpClient = new MockHttpClient(function (string $method, string $url, array $options) {
+            $this->assertSame('GET', $method);
+            $this->assertStringStartsWith('https://huggingface.co/api/models', $url);
+            $this->assertArrayHasKey('query', $options);
+            $this->assertSame('test-provider', $options['query']['inference_provider']);
+            $this->assertSame('text-generation', $options['query']['pipeline_tag']);
+
+            return new JsonMockResponse([]);
+        });
+
+        $apiClient = new ApiClient($httpClient);
+        $apiClient->models('test-provider', 'text-generation');
+    }
+
+    #[TestDox('Sends correct HTTP request with null provider and task parameters')]
+    public function testModelsRequestParametersWithNullValues()
+    {
+        $httpClient = new MockHttpClient(function (string $method, string $url, array $options) {
+            $this->assertSame('GET', $method);
+            $this->assertStringStartsWith('https://huggingface.co/api/models', $url);
+            $this->assertArrayHasKey('query', $options);
+            $this->assertNull($options['query']['inference_provider']);
+            $this->assertNull($options['query']['pipeline_tag']);
+
+            return new JsonMockResponse([]);
+        });
+
+        $apiClient = new ApiClient($httpClient);
+        $apiClient->models(null, null);
+    }
```

Comment:
Does this make sense @chr-hertel or should we validate, that at least one needs to be set?